vcu/Src/drive_by_wire_mock.c

```diff
@@ -440,6 +440,75 @@ static const CLI_Command_Definition_t mcInitCommandDefinition =
     0 /* Number of parameters */
 };
 
+
+BaseType_t setInverterParameter(char *writeBuffer, size_t writeBufferLength, const char *commandString){
+    uint16_t parameterAddress;
+    uint16_t newValue;
+    BaseType_t status;
+    uint16_t readValue = 0;
+    uint32_t currentState = fsmGetState(&fsmHandle); //returns an index
+    BaseType_t paramLen;
+
+    const char *addressParam = FreeRTOS_CLIGetParameter(commandString, 1, &paramLen);
+    const char *valueParam = FreeRTOS_CLIGetParameter(commandString, 2, &paramLen);
+    
+    sscanf(addressParam, "%hu", &parameterAddress);
+    sscanf(valueParam, "%hu", &newValue);
+
+    if (HV_Power_State == HV_Power_State_On){ 
+        COMMAND_OUTPUT("Error: Car is at high voltage. Operation not permitted\n");
+        return pdFALSE;
+    }
+    else if (currentState == STATE_EM_Enable){ 
+        COMMAND_OUTPUT("Error: Car is in drive mode. Operation not permitted\n");
+        return pdFALSE;
+    } else{
+
+
+        status = mcWriteParamCommand(parameterAddress, newValue); 
+        if (status == HAL_OK){
+            COMMAND_OUTPUT("Parameter written successfully. \n");
+        }
+        if (status != HAL_OK){
+            COMMAND_OUTPUT("Failed to write parameter message at address %u\n", parameterAddress);
+            return pdFALSE;
+        }
+
+        HAL_Delay(1000);
+
+        status = mcReadParamCommand(parameterAddress, readValue);
+        if (status == HAL_OK){
+            COMMAND_OUTPUT("Parameter read successfully. \n");
+
+        if (newValue == readValue){
+            COMMAND_OUTPUT("Parameter verification successful: %u == %u\n", newValue, readValue);
+            return pdTRUE;
+        }
+        if (newValue != readValue){
+            COMMAND_OUTPUT("Parameter verification unsuccessful: %u == %u\n", newValue, readValue);
+            return pdFALSE;
+        }
+    }
+    if (status != HAL_OK){
```

Comment:
your formatting is off here, if you're inside the else statement you should make sure your indenting Is okay typically everything inside should be indented. although I want you to get rid of the else statement like I mentioned previously cause it serves no purpose